WalletWasabi/Crypto/ZeroKnowledge/LinearRelation/Knowledge.cs

```diff
@@ -1,37 +0,0 @@
-using NBitcoin.Secp256k1;
-using System.Collections.Generic;
-using System.Linq;
-using WalletWasabi.Helpers;
-
-namespace WalletWasabi.Crypto.ZeroKnowledge.LinearRelation
-{
-	public class Knowledge
-	{
-		public Knowledge(Statement statement, ScalarVector witness)
-		{
-			Guard.NotNull(nameof(statement), statement);
-			Guard.NotNullOrEmpty(nameof(witness), witness);
-			Guard.True($"{nameof(witness)} size does not match {nameof(statement)}.{nameof(statement.Equations)}", witness.Count() == statement.Equations.First().Generators.Count());
-
-			// don't try to prove something which isn't true
-			foreach (var equation in statement.Equations)
-			{
-				Guard.True($"{nameof(witness)} is not solution of the {nameof(equation)}", equation.VerifySolution(witness));
-			}
```

Comment:
these guards should be preserved in `Statement.RespondToChallenge` if we go with it